Reference/Notifications/UmbracoApplicationLifetime-Notifications/index.md

```diff
@@ -0,0 +1,127 @@
+---
+versionFrom: 9.3.0
+verified-against: 9.3.0
+meta-title: Umbraco Application Lifetime Notifications
+meta.Description: Represents an Umbraco application lifetime (starting, started, stopping, stopped) notification
+state: complete
+update-links: true
+---
+
+# Umbraco Application Lifetime Notifications
+
+Umbraco application lifetime notifications are published for the starting, started, stopping, and stopped events of the Umbraco runtime. These events implement the `IUmbracoApplicationLifetimeNotification` interface that contain a single `IsRestarting` property.
+
+An Umbraco application is restarted after an install or upgrade has been completed, so you can use this property to prevent running code twice (on initial boot and restart). To prevent running code when the application is in the install or upgrade state, inject an `IRuntimeState` instance in your notification and inspect the `Level` property instead.
```

Comment:
```suggestion
A Umbraco application is restarted after an install or upgrade has been completed, so you can use this property to prevent running code twice (on initial boot and restart). To prevent running code when the application is in the install or upgrade state, inject an `IRuntimeState` instance in your notification and inspect the `Level` property instead.
```